packages/-ember-data/tests/unit/adapters/rest-adapter/build-query-test.js

```diff
@@ -27,4 +27,26 @@ module('unit/adapters/rest-adapter/build-query - building queries', function() {
 
     assert.deepEqual(query, { include: 'comments' }, 'query includes `include`');
   });
+
+  test('buildQuery() returns query with `fields` from snapshot', function(assert) {
+    const adapter = DS.RESTAdapter.create();
+    const snapshotStub = { fields: '[post]=name' };
+
+    const query = adapter.buildQuery(snapshotStub);
+
+    assert.deepEqual(query, { fields: '[post]=name' }, 'query includes `fields`');
+  });
+
+  test('buildQuery() returns query with `fields` and `include` from snapshot', function(assert) {
+    const adapter = DS.RESTAdapter.create();
+    const snapshotStub = { fields: '[post]=name&[comments]=title', include: 'comments' };
```

Comment:
In my app, when I pass in fields to the `query` funciton, it is an object like this:
```
{
fields: {
  post: 'name',
  comments: 'title'
}
include: 'comments'
}
```

Can I assume that this will work with `findRecord` as well?

Reply:
Can can certainly make this work as well!  Good idea

Reply:
Looks like we properly handle serializing the query params to the url!  Thanks for the comment.  Added a test here -

https://github.com/emberjs/data/pull/7234/commits/79e2507dc9541f8529281ee28dca9526721203cd